dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,277 @@
+                else render_s3_table_function(
+                    bucket_http_url,
+                    file_format=file_extension,  # type: ignore[arg-type]
+                )
+            )
+        elif bucket_scheme in ("az", "abfs"):
+            if isinstance(staging_credentials, AzureCredentialsWithoutDefaults):
+                # Authenticated access.
+                account_name = staging_credentials.azure_storage_account_name
+                storage_account_url = (
+                    f"{staging_credentials.azure_storage_account_name}.blob.core.windows.net"
+                )
+                account_key = staging_credentials.azure_storage_sas_token
+                container_name = bucket_url.netloc
+                blobpath = bucket_url.path
+
+                format_mapping = {"jsonl": "JSONEachRow", "parquet": "Parquet"}
+                clickhouse_format = format_mapping[file_extension]
+
+                table_function = (
+                    f"azureBlobStorage('{storage_account_url}','{container_name}','{ blobpath }','{ account_name }','{ account_key }','{ clickhouse_format}')"
+                )
+
+            else:
+                # Unsigned access.
+                raise NotImplementedError(
+                    "Unsigned Azure Blob Storage access from Clickhouse isn't supported as yet."
+                )
+
+        with client.begin_transaction():
+            client.execute_sql(
+                f"""INSERT INTO {qualified_table_name} SELECT * FROM {table_function}"""
+            )
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class ClickhouseClient(SqlJobClientWithStaging, SupportsStagingDestination):
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(
+        self,
+        schema: Schema,
+        config: ClickhouseClientConfiguration,
+    ) -> None:
+        # TODO: There are no schemas in Clickhouse. No point in having schemas, only dataset names and table names for example "dataset1_mytable".
+        self.sql_client: ClickhouseSqlClient = ClickhouseSqlClient(
+            config.normalize_dataset_name(schema), config.credentials
+        )
+        super().__init__(schema, config, self.sql_client)
+        self.config: ClickhouseClientConfiguration = config
+        self.active_hints = deepcopy(HINT_TO_CLICKHOUSE_ATTR)
+        self.type_mapper = ClickhouseTypeMapper(self.capabilities)
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        return super().start_file_load(table, file_path, load_id) or ClickhouseLoadJob(
+            file_path,
+            table["name"],
+            self.sql_client,
+            staging_credentials=(
+                self.config.staging_config.credentials if self.config.staging_config else None
+            ),
+        )
+
+    def _get_table_update_sql(
```

Comment:
just for me to understand, clickhouse does not have any schemas at all, you just define the table and an engine and then the schema is whatever is found in the files loaded to that table?

Reply:
Correct, the schema is just the prefix